usaspending_api/search/tests/unit/test_spending_by_award_type_subawards.py

```diff
@@ -31,9 +31,9 @@ def test_spending_by_award_subawards_fail(client, refresh_matviews):
     resp = client.post(
         "/api/v2/search/spending_by_award",
         content_type="application/json",
-        data=json.dumps({"fields": ["Sub-Award ID"], "filters": {"award_type_codes": ["06"]}, "subawards": True}),
+        data=json.dumps({"fields": ["Sub-Award ID"], "filters": {"award_type_codes": ["X"]}, "subawards": True}),
     )
-    assert resp.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+    assert resp.status_code == status.HTTP_400_BAD_REQUEST
```

Comment:
How come?

Reply:
The 422 was thrown when the award type code was recognized but not allowed (i.e., direct payments, loans, etc.) Now that we allow those, this error wasn't being thrown, so I changed the test to be for a non-existent type code, which throws a 400 error.

Reply:
A lot of stuff could throw a 400. Sure we couldn't have it be an IDV type or something and keep that 422?

Reply:
We did have a user issue about throwing a 500 when really all that was wrong was the user input, and this could lead a future user to the same issue.

Reply:
IDVs are an accepted type, so it wouldn't give a 422. Anything outside of the actual type codes gets caught in a separate check, I assume it's tinyshield. The API response  is pretty clear about what the issue is. `{
    "detail": "Field 'filters|award_type_codes' is outside valid values ['10', 'B', 'IDV_A', '07', '04', 'IDV_B_B', '08', '11', 'D', 'IDV_B_C', '09', 'IDV_E', 'C', 'IDV_B_A', '06', 'A', '05', 'IDV_B', '03', 'IDV_D', '02', 'IDV_C', 'no intersection']"
}`